src/ml_tooling/plots/target_feature_distribution.py

```diff
@@ -0,0 +1,72 @@
+from matplotlib import pyplot as plt
+from matplotlib.axes import Axes
+import pandas as pd
+import numpy as np
+
+from ml_tooling.utils import DataType
+from scipy import stats
+
+
+def target_feature_distribution(
+    data: pd.DataFrame,
+    targets: DataType,
+    features: DataType,
+    title: str = None,
+    labels: list = None,
+    median: bool = False,
+) -> Axes:
+
+    """
+    Creates two plots, one which compares the mean or median
+    of a target based on the given features and another which shows
+    the distribution of the target based on the given features
+
+    :param data:
+        DataFrame containing the data
+
+    :param targets:
+        The name of the column containing the targets
+
+    :param features:
+        The name of the column containing the features
+
+    :param title:
+        Title for plot
+
+    :param labels:
+        Pass custom list of labels
+
+    :param median:
+        Whether to compare using median or mean, true for median
+
+    :return:
+        matplotlib.Axes
+    """
+
+    title = f"Survival Distributions: {targets}" if title is None else title
+    fig, axs = plt.subplots(nrows=1, ncols=2)
+    fig.suptitle(title)
+
+    ax1df = data.groupby(by=targets)[features]
+    std = ax1df.std()
+    count = ax1df.count()
+    yerr = std / np.sqrt(count) * stats.t.ppf(1 - 0.05 / 2, count - 1)
```

Comment:
This only works if the data is normally distributed - a generalized solution would be to implement bootstrapping.